charts/gafaelfawr/templates/redis-networkpolicy.yaml

```diff
@@ -1,19 +1,23 @@
 apiVersion: networking.k8s.io/v1
 kind: NetworkPolicy
 metadata:
-  name: {{ template "helpers.fullname" . }}-redis-networkpolicy
+  name: {{ template "gafaelfawr.fullname" . }}-redis-networkpolicy
+  labels:
+    {{- include "gafaelfawr.labels" . | nindent 4 }}
 spec:
   podSelector:
     matchLabels:
-      app: {{ template "helpers.fullname" . }}-redis
+      {{- include "gafaelfawr.selectorLabels" . | nindent 6 }}
+      app.kubernetes.io/component: "redis"
   policyTypes:
     - Ingress
     - Egress
   ingress:
     - from:
         - podSelector:
             matchLabels:
-              name: {{ template "helpers.fullname" . }}
+              {{- include "gafaelfawr.selectorLabels" . | nindent 14 }}
+              app.kubernetes.io/component: "frontend"
```

Comment:
not sure exactly what this means? frontend?  is this like a DMZ zone or something?

Reply:
The default chart selector labels are only unique within the chart, so if there's more than one deployment in the chart (or in this case a deployment and a statefulset), you have to add an additional label to distinguish the pods. Using this component label is the Helm best practices recommendation. The value is arbitrary; it just needs to be different between the web server and Redis. I picked frontend just because within the Gafaelfawr deployment the web service is the "front end."

Reply:
but this is the redis network policy?  Should this be redis?

Reply:
It's the ingress rule that lets the web server talk to the Redis server, so it has to designate the web server. (The bit that matches the redis server is up above, in the previous hunk.)

Reply:
Ah okay thanks

Reply:
Actually, maybe we should document each of these rules with a comment in english on what they are?  I think jupyter's charts do that and it's pretty helpful

Reply:
Good call, I'll add that to my to-do list.